bangc-ops/mlu_op.h

```diff
@@ -1610,6 +1610,180 @@ mluOpStatus_t MLUOP_WIN_API mluOpGetPolyNmsWorkspaceSize(
     mluOpHandle_t handle, const mluOpTensorDescriptor_t boxes_desc,
     size_t *size);
 
+/*!
+ *  @brief Gets extra space size that is needed in poly_nms operation.
+ *
+ *  @param[in] handle
+ *    Input. Handle to a MLUOP context that is used to manage MLU devices
+ *    and queues in the psroipool_forward operation.
+ *  @param[in] scores_desc
+ *    Input. The descriptor of the scores tensor. For detailed information,
+ *    see ::mluOpTensorDescriptor_t. 
+ *  @param[out] size
+ *    Output. A host pointer to the returned size of extra space in bytes.
+ *  @par Return
+ *  - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM
+ */
+mluOpStatus_t MLUOP_WIN_API mluOpGetGenerateProposalsV2WorkspaceSize(
+    mluOpHandle_t handle, const mluOpTensorDescriptor_t scores_desc,
+    size_t *size);
+
+/*!
+ *  @brief This operator is the second version of generate_proposals op to
+ *    generate bounding box proposals for Faster RCNN.
+ *    The proposals are generated for a list of images based on image
+ *    score 'Scores', bounding box regression result 'BboxDeltas' as
+ *    well as predefined bounding box shapes 'anchors'. Greedy non-maximum
+ *    suppression is applied to generate the final bounding boxes.
```

Comment:
Generates bounding box proposals for faster RCNN.  This operator is the second version of generate_proposals.

The proposals are generated for a list of images based on image
score 'Scores', bounding box regression result 'BboxDeltas' as
well as predefined bounding box shapes 'anchors'. Greedy non-maximum
suppression is applied to generate the final bounding boxes.